content/Gems You Want To Have In Your Rails API - Juan Diego Rosales.md

```diff
@@ -0,0 +1,44 @@
+
+If you need a custom page, you can have Active Admin render your own partial and you will probably need to make changes in the controller actions. This is not an easy task, but none of the other gems offer anything better.
+
+For some people, one of the downsides could be the design, since Active Admin does not come with a powerful CSS framework out of the box. Luckily, there are many complementary gems that provide a better looking theme, for example [Arctic Admin](https://github.com/cprodhomme/arctic_admin).
+
+## Background jobs
+On the background processing front, I would definitely recommend starting with [Delayed Job](https://github.com/collectiveidea/delayed_job). This is because it easily integrates with Rails, has minimal dependencies, resulting in a fast setup, and it offers a very stable and reliable processing.
+
+If your application scales, you will need a framework which delivers performace and in that case I would recommend switching to [Sidekiq](https://github.com/mperham/sidekiq). Changing the framework should be an easy task thanks to ActiveJob and you will get awesome benefits from this. Sidekiq is known for its very fast speed, scalability and multithreading potential. By running in memory with Redis, fetching and saving data is much faster than completely relying on the database like Delayed Job, and what's more, you can have a dashboard to monitor the status of the processes.
+
+## Testing
+Although it may take some time to get used to the DSL, here at Rootstrap we choose [RSpec](https://github.com/rspec/rspec-rails) for testing. The main reason for this is that it encourages human readable tests, which ends up helping a lot in the development process.
+
+When writing tests you always need to set up database records to create the context, and for this we like to use factories with [Factory Bot](https://github.com/thoughtbot/factory_bot). This gem offers great flexibility, allowing the developer to easily create different scenarios, but also improves the readability as you can easily know what is going on in the test.
+
+Last but not least, if your API relies on third party services you will need stubbing on the external HTTP requests and for this I recommend using [Webmock](https://github.com/bblimke/webmock). If you ask me, I found this gem much simpler and flexible than VCR, for example.
+
+## Linting
+Finally, every developer wants a good quality codebase and in that sense linters are essential. Here's a list of gems to consider:
+
+* [RuboCop](https://github.com/rubocop-hq/rubocop-rails)
+* [Rails Best Practices](https://github.com/flyerhzm/rails_best_practices)
+* [Reek](https://github.com/troessner/reek) for code smells
+* [Bullet](https://github.com/flyerhzm/bullet) for query optimization
+* [i18n-tasks](https://github.com/glebm/i18n-tasks) for translations linting
```

Comment:
whats the difference with our i18n_linter ?

Reply:
I recommended the one that was in our Rails API base, and I have worked with it as well. I didn't know about that gem of ours...

Reply:
As we discussed go ahead with your proposed gem 👍